pkg/api/scylla/v1alpha1/types_operatorconfig.go

```diff
@@ -5,12 +5,49 @@ package v1alpha1
 import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 type ScyllaOperatorConfigSpec struct {
-	// scyllaUtilsImage is a Scylla image used for running scylla utilities.
-	// +kubebuilder:validation:MinLength=1
+	// scyllaUtilsImage is a ScyllaDB image used for running ScyllaDB utilities.
 	ScyllaUtilsImage string `json:"scyllaUtilsImage"`
+
+	// unsupportedBashToolsImageOverride allows to adjust a generic Bash image with extra tools used by the operator
+	// for auxiliary purposes.
+	// Setting this field renders your cluster unsupported. Use at your own risk.
+	// +optional
+	UnsupportedBashToolsImageOverride *string `json:"unsupportedBashToolsImageOverride,omitempty"`
```

Comment:
`Unsupported` prefix is confusing. In the past, users of fields prefixed with it (unsupportedOptions in NodeConfig mounts) didn't understand whether this field works, what it does, because of this prefix. I think it's a mistake to use it. 

I think it's enough to make it explicit in the field description, that using it should be restricted to advanced users knowing that they are doing. We can explain consequences of changing it there.
Looking at Kubernetes resources, there're none having this confusing prefix, yet there're field which users shouldn't set (Job selector for example).

Reply:
users don't read the docs and the field name is the only thing that they see so I'd stick with it. it's very much the same like UnsupportedScyllaDBArgs

Reply:
Then why we are getting complains about quality of documentation? They do, and it's the right place to write longer descriptions. 

We don't have UnsupportedScyllaDBArgs in new API, we have AdditionalScyllaDBArgs + description.

Reply:
> Then why we are getting complains about quality of documentation? They do

This is a probabilistic case. Just a few users/customers, who don't, are enough and manage to build something on top of the unsupported options and you'll get all the fires when it finally breaks no matter the docs said unsupported. At OpenShift, we have been there, see e.g.
https://github.com/openshift/api/blob/01b3675ba7b364e312ac5da6e632251fbaefdda0/operator/v1/types_ingress.go#L254
https://github.com/search?q=repo%3Aopenshift%2Fapi+unsupported&type=code&p=2
It's better to be safe then sorry. I don't intent to get paged over that.

Reply:
I'm sure they will build on top of it anyway. Having confusing prefix or not is not going to stop them. That's why we should improve documentation and user understanding. Having confusing prefix doesn't help with that.

Reply:
> I'm sure they will build on top of it anyway. Having confusing prefix or not is not going to stop them.

I think this is more about having a stronger case for not carting about such cases than it is about preventing it entirely. I do however agree with @zimnx that `Unsupported` prefix is confusing, I myself didn't understand what `unsupportedOptions` in NodeConfig were supposed to stand for at first, but that's something that actually encourages you to go and check, instead of just using it. Perhaps we could find a better prefix, but I don't have a better idea myself. Between `unsupported` and dropping the prefix altogether, I'd go with the former.

Reply:
> That's why we should improve documentation and user understanding. Having confusing prefix doesn't help with that.

It does, the prefix is meant for people who don't reach the docs. I have a hard time undestanding how enhancing the docs helps the group of people who don't read it.

Reply:
We shouldn't optimize for slackers, it's entirely their fault they don't read the documentation.
I can imagine someone trying to deploy 1.0 Scylla without checking supported versions - part of the documentation. Following your logic, shouldn't we add confusing prefix to `version` field to encourage them to read the docs? 

Sounds like a security by obscurity but for API.

Reply:
Without any new input I am afraid we have to agree to disagree on this one.